src/commands/hop.tsx

```diff
@@ -23,30 +26,40 @@ function Hop() {
         void getLocalBranches();
     }, [git, setAllBranches, setCurrentBranch]);
 
-    const handleSelect = (item: { label: string; value: string }) => {
-        const updateCurrentBranch = async () => {
-            const { current } = await git.branchLocal();
-            setNewBranch(current);
-        };
+    const handleSelect = useCallback(
+        (item: { label: string; value: string }) => {
+            const updateCurrentBranch = async () => {
+                const { current } = await git.branchLocal();
+                setNewBranch(current);
+            };
 
-        git.checkout(item.value)
-            .then(async () => {
-                return updateCurrentBranch();
-            })
-            .catch((error: Error) => {
-                setError(error);
-            })
-            .finally(() => {
-                exit();
-            });
-    };
+            git.checkout(item.value)
+                .then(() => {
+                    void updateCurrentBranch();
+                })
+                .catch((error: Error) => {
+                    setError(error);
+                })
+                .finally(() => {
+                    exit();
+                });
+        },
+        [git, setNewBranch, setError, exit]
+    );
 
     const items = useMemo(() => {
         if (allBranches.length === 1) {
             return [{ label: currentBranch, value: currentBranch }];
         }
         return allBranches
             .filter((branch) => branch !== currentBranch)
+            .filter((branch) =>
```

Comment:
So there's a bit of a bug where if the filtered branches have a length of only 1, the `hopped to <branch_name>` message does not show up anymore. Might be because of re-renders and it catching this early return case. It is consistently reproducible